homeassistant/components/lovelace/__init__.py

```diff
@@ -118,10 +119,12 @@ def save_yaml(fname: str, data: JSON_TYPE):
     yaml.indent(sequence=4, offset=2)
     tmp_fname = fname + "__TEMP__"
     try:
-        with open(os.open(tmp_fname, O_WRONLY | O_CREAT | O_TRUNC, 0o644),
+        file_stat = os.stat(fname)
+        with open(os.open(tmp_fname, O_WRONLY | O_CREAT | O_TRUNC, file_stat[ST_MODE]),
                   'w', encoding='utf-8') as temp_file:
             yaml.dump(data, temp_file)
         os.replace(tmp_fname, fname)
+        os.chown(fname, file_stat[ST_UID], file_stat[ST_GID])
```

Comment:
Not sure what this will do on Windows?

Reply:
I don't know if we should do this, as it might cause weird issues if user is not allowed to chown it to the other user, which I expect is most of the times.